provider/oauth2.go

```diff
@@ -235,9 +247,108 @@ func (p Oauth2Handler) LogoutHandler(w http.ResponseWriter, r *http.Request) {
 	p.JwtService.Reset(w)
 }
 
+func (p Oauth2Handler) loadUserFromIDToken(tok *oauth2.Token) (UserData, []byte, error) {
+	idToken, ok := tok.Extra("id_token").(string)
+	if !ok || idToken == "" {
+		return nil, nil, fmt.Errorf("id_token not found")
+	}
+
+	if p.keyfunc == nil {
+		err := p.tryInitJWKSKeyfunc()
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "can't load JWKS keys")
+		}
+	}
+
+	claims := jwt.MapClaims{}
+	parser := jwt.Parser{
+		// claims validation is not considering clock skew and randomly failing with iat validation
+		// nbf and exp are validated below
+		SkipClaimsValidation: true,
```

Comment:
I get the idea and the need, but still makes me worry. Maybe there is a way to do it safer somehow without skip validation?

Reply:
Upgrading to `jwt/v4` should solve this issue as well, they added clock skew recently.